### PR TITLE
Bugfix and new options related to history variable predicates

### PR DESCRIPTION
**Pull request by makaimann**

### CODE REVIEW

**Comment by makaimann** [comment]
I thought we should merge these fixes/improvements. I haven't looked carefully over all the changes -- there's lots of changed files because the mathsat and ic3ia update. So I kept it as a draft PR for now.

File: ic3ia/utils.cpp [hunk 1]
```diff
@@ -82,11 +82,11 @@ msat_config get_config(ModelGeneration model, bool interpolation)
         // interpolation for BV requires the lazy solver
         OPT_("theory.bv.bit_blast_mode", "0");
         OPT_("theory.bv.eager", "false");
-	// This option is a hack to handle interpolation failure in the
-	// case of QF_UFLIA
-	OPT_("theory.la.interpolation_laz_split_mixed_eq", "false");
+        // This option is a hack to handle interpolation failure in the
+        // case of QF_UFLIA
+        OPT_("theory.la.interpolation_laz_split_mixed_eq", "false");
```
Comment (ahmed-irfan) [inline ic3ia/utils.cpp:87]: let's remove this option

File: prophic3/array_abstractor.cpp [hunk 2]
```diff
@@ -64,31 +64,9 @@ void ArrayAbstractor::do_abstraction()
   msat_term new_prop = abstract(conc_ts_.prop());
 
   // need to promote inputs that occur in new_init / new_prop to states
-  // this includes the created witnesses which don't explicitly appear
+  // this includes the created witnesses which are used as indices and should be states
```
Comment (ahmed-irfan) [inline prophic3/array_abstractor.cpp:67]: update this comment. Now witnesses are not states

File: ic3ia/utils.cpp (hunk 1 above)
Comment (makaimann) [reply ic3ia/utils.cpp:87]: Just for future reference: we decided offline to keep this option for now and change it in a separate PR

**Pull request merged**